sources/ClangSharp/Extensions/CXTranslationUnit.cs

```diff
@@ -1,28 +1,84 @@
-
-        public static CXErrorCode Create(CXIndex index, string astFileName, out CXTranslationUnit translationUnit) => clang.createTranslationUnit2(index, astFileName, out translationUnit);
-
-        public static CXTranslationUnit CreateFromSourceFile(CXIndex index, string sourceFileName, string[] commandLineArgs, CXUnsavedFile[] unsavedFiles) => clang.createTranslationUnitFromSourceFile(index, sourceFileName, commandLineArgs.Length, commandLineArgs, (uint)unsavedFiles.Length, unsavedFiles);
-
-        public static CXTranslationUnit Parse(CXIndex index, string sourceFileName, string[] commandLineArgs, CXUnsavedFile[] unsavedFiles, CXTranslationUnit_Flags options) => clang.parseTranslationUnit(index, sourceFileName, commandLineArgs, commandLineArgs.Length, unsavedFiles, (uint)unsavedFiles.Length, (uint)options);
-
-        public static CXErrorCode Parse(CXIndex index, string sourceFileName, string[] commandLineArgs, CXUnsavedFile[] unsavedFiles, CXTranslationUnit_Flags options, out CXTranslationUnit translationUnit) => clang.parseTranslationUnit2(index, sourceFileName, commandLineArgs, commandLineArgs.Length, unsavedFiles, (uint)unsavedFiles.Length, (uint)options, out translationUnit);
-
-        public static CXErrorCode ParseFullArgv(CXIndex index, string sourceFileName, string[] commandLineArgs, CXUnsavedFile[] unsavedFiles, CXTranslationUnit_Flags options, out CXTranslationUnit translationUnit) => clang.parseTranslationUnit2FullArgv(index, sourceFileName, commandLineArgs, commandLineArgs.Length, unsavedFiles, (uint)unsavedFiles.Length, (uint)options, out translationUnit);
+        public static CXTranslationUnit Create(CXIndex index, string astFileName)
+        {
+            using (var marshaledAstFileName = new MarshaledString(astFileName))
+            {
+                return clang.createTranslationUnit(index, marshaledAstFileName);
+            }
+        }
+
+        public static CXErrorCode Create(CXIndex index, string astFileName, out CXTranslationUnit translationUnit)
+        {
+            using (var marshaledAstFileName = new MarshaledString(astFileName))
+            fixed (CXTranslationUnit* pTranslationUnit = &translationUnit)
+            {
+                return clang.createTranslationUnit2(index, marshaledAstFileName, (CXTranslationUnitImpl**)pTranslationUnit);
+            }
+        }
+
+        public static CXTranslationUnit CreateFromSourceFile(CXIndex index, string sourceFileName, string[] commandLineArgs, CXUnsavedFile[] unsavedFiles)
+        {
+            using (var marshaledSourceFileName = new MarshaledString(sourceFileName))
+            using (var marshaledCommandLineArgs = new MarshaledStringArray(commandLineArgs))
+            fixed (CXUnsavedFile* pUnsavedFiles = unsavedFiles)
+            {
+                var pCommandLineArgs = stackalloc sbyte*[marshaledCommandLineArgs.Count];
```

Comment:
Marshaling string arrays is currently a bit more complex. I had tried to allocate a `sbyte**` and then fill it directly (to avoid needing to allocate a `sbyte*[]`), but that was having problems and I decided to resolve it later.